sort_names.py

```diff
@@ -1,95 +1,188 @@
-import csv
+import csv  # work with .csv files
 import sys
 import time
+import urllib.request
+from collections import defaultdict
 from fileinput import filename
-from deepface import DeepFace
 
+# modules
+import filters
+import inputs
 
 filename = sys.argv[1]
 
 # ---- INPUT PARAMETERS ----
-limit_age = 25  # not younger than
-races = [
-    "black",
-    "white",
-    "asian",
-]  # enter necessary races. List of possible races - ['asian', 'indian', 'black', 'white', 'middle eastern', 'latino hispanic']
-forbidden_word = "NFT"  # a forbidden word in the current position of person
-black_list = "Indusi.csv"  # Filter by name. CSV file with two columns - First_name and Second_name
+LIMIT_AGE = 25  # not younger than
+# enter necessary races. List of possible races - ['asian', 'indian', 'black', 'white', 'middle eastern', 'latino hispanic']
+races = ["asian", "black", "white"]
+DEL_PEOPLE_WITHOUT_AVATAR = False  # past 'True' or 'False'
 
 
-def get_short_list(filename, first_names, last_names):
+def conditions(row, dict_races, first_names, last_names, positions):
+    """Check all conditions and if row is correct - return True, else - False.
+    At the same time returns a tuple with counters
+    """
+    if filters.name_filter(row[24], first_names):
+        if row[25] not in last_names:
+            if filters.position_filter(row[43], positions):
+                if (row[30] == "") and DEL_PEOPLE_WITHOUT_AVATAR:
+                    return False, (
+                        "count_name",
+                        "count_last",
+                        "count_pos",
+                        "count_no_link",
+                    )
+                elif (row[30] == "") and not DEL_PEOPLE_WITHOUT_AVATAR:
+                    return True, (
+                        "count_name",
+                        "count_last",
+                        "count_pos",
+                        "count_no_link",
+                        "count",
+                    )
+                else:
```

Comment:
Need to think how to avoid multiple else clauses.
Let's try to have a list of conditions.
[cond1, cond2, cond3] and use short circuiting on going through them. Or filter, or all()

Reply:
I see a problem - if I create this list, all these conditions should get true/False values (problem with the last condition - deepFace filter - it is too long). Now if the first, second or third condition gets False - the function stops the execution and return something

Reply:
@OleksandrNovitskyi make last condition a function that returns true/false

Reply:
> Now if the first, second or third condition gets False - the function stops the execution and return something

this is how short-circuiting works.

Reply:
I can't rewrite it(( it is too complicated code to me(

Reply:
Sorry for bad explanation, let me rewrite it as example.